api/maven-api-core/src/main/java/org/apache/maven/api/feature/Features.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.api.feature;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+
+import org.apache.maven.api.Session;
+import org.apache.maven.api.annotations.Nonnull;
+
+/**
+ * Centralized class for feature information
+ *
+ * @since 4.0.0
+ */
+public final class Features {
+
+    public static final String BUILDCONSUMER = "maven.experimental.buildconsumer";
+
+    public static final String XINCLUDE = "maven.experimental.xinclude";
+
+    public static final String TRUE = "true";
+
+    private Features() {}
+
+    @Nonnull
+    public static boolean buildConsumer(@Nonnull Properties userProperties) {
+        return TRUE.equals(Objects.requireNonNull(userProperties).getOrDefault(BUILDCONSUMER, TRUE));
```

Comment:
In fact, reading this I realize the constant actively obscures the type, which is obvious with a string literal.